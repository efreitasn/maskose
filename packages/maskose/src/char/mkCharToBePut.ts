import { MaskCharPredicateFn } from '.';

const MASK_CHAR_TO_BE_PUT_TYPE = 'MASK_CHAR_TO_BE_PUT_TYPE';

export type MaskCharToBePutObj = {
  type: typeof MASK_CHAR_TO_BE_PUT_TYPE;
  char: string;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character that's expected to be put in the mask when formatting instead of being a character
 * expected to be in the value to be masked. For example, the dash (-) in a phone number mask is
 * such a character
 * @param char Character to be put in the mask
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharToBePut(char: string, predicateFn?: MaskCharPredicateFn): MaskCharToBePutObj {
  return {
    type: MASK_CHAR_TO_BE_PUT_TYPE,
    char,
    predicateFn
  };
}