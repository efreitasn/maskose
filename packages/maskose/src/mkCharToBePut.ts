import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_TO_BE_PUT_TYPE = 'MASK_CHAR_TO_BE_PUT_TYPE';

export default function mkCharToBePut(char: string, predicateFn?: MaskCharPredicateFn) {
  return {
    type: MASK_CHAR_TO_BE_PUT_TYPE,
    char,
    predicateFn
  };
}