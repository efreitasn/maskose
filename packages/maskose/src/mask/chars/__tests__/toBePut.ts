import mkCharToBePut from '../toBePut';

it('should match the object returned by mkCharToBePut()', () => {
  const result = mkCharToBePut('(');

  expect(result).toMatchSnapshot();
});

it('should throw an error when an string with length > 1 is provided', () => {
  expect(() => mkCharToBePut('))')).toThrow();
  expect(() => mkCharToBePut('str')).toThrow();
});