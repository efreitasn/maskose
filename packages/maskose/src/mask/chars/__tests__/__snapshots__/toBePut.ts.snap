// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkCharToBePut() 1`] = `
Object {
  "char": "(",
  "maskedValueLengthConditions": Array [],
  "regExp": /\\\\\\(/,
  "repetitions": 1,
  "type": "MASKOSE_CHAR_TO_BE_PUT_TYPE",
  "valueToBeMaskedLengthConditions": Array [],
}
`;
